playground/DatabaseMigration/DatabaseMigration.ApiService/Program.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.EntityFrameworkCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.AddServiceDefaults();
+
+builder.AddSqlServerDbContext<MyDb1Context>("db1", configureDbContextOptions: options =>
+{
+    options.UseSqlServer(sqlServerOptions =>
+    {
+        sqlServerOptions.MigrationsAssembly("DatabaseMigration.ApiModel");
```

Comment:
To follow the migrations project pattern you also need to move the migrations to `DatabaseMigration.MigrationService` as they are not used during production app execution.

Reply:
I get this error:

> Could not load assembly 'DatabaseMigration.MigrationService'. Ensure it is referenced by the startup project 'DatabaseMigration.ApiService'.

I don't want the ApiService project (production web app) to reference MigrationService (worker app).

Reply:
That's fine, you should use `DatabaseMigration.AppHost` as the startup project. And `DatabaseMigration.MigrationService` would be the target/migration project. See [Using a Separate Migrations Project](https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/projects?tabs=dotnet-core-cli)

Reply:
`AppHost` doesn't know about the DB context. It doesn't execute `ApiService` startup.

Reply:
I see. While the exception message says it needs to be referenced, in reality it just needs to be a development dependency. Unfortunately, this does not work currently

```
    <ProjectReference Include="../DatabaseMigration.MigrationService/DatabaseMigration.MigrationService.csproj">
      <PrivateAssets>all</PrivateAssets>
      <ExcludeAssets>compile</ExcludeAssets>
    </ProjectReference>
```

due to https://github.com/NuGet/Home/issues/10907

Reply:
There is discussion about making the migration available in the app host here: https://github.com/dotnet/aspire/issues/2090. That change could allow the pattern you want.

It sounds like the benefit is not having the migration in production code. I think it's a nice to have, but not a blocker.

Reply:
Migrations in apphost is nice in theory but dependency conflicts are a big concern.